src/lua/datetime.lua

```diff
@@ -1085,16 +1089,26 @@ local function datetime_set(self, obj)
         end
         local sec_int, fraction
         sec_int, fraction = math_modf(ts)
+        if fraction < 0 then
```

Comment:
add a comment here with explanation

Reply:
Added. (With simple changes to be consistent in assign operations)

```diff
--- a/src/lua/datetime.lua
+++ b/src/lua/datetime.lua
@@ -588,9 +588,12 @@ local function datetime_new(obj)
         end
         local fraction
         s, fraction = math_modf(ts)
+        -- In case of negative fraction part we should
+        -- make it positive at the expense of the integer part.
+        -- Code below expects that "nsec" value is always positive.
         if fraction < 0 then
             s = s - 1
-            fraction = 1 + fraction
+            fraction = fraction + 1
         end
         -- if there are separate nsec, usec, or msec provided then
         -- timestamp should be integer
@@ -1089,9 +1092,12 @@ local function datetime_set(self, obj)
         end
         local sec_int, fraction
         sec_int, fraction = math_modf(ts)
+        -- In case of negative fraction part we should
+        -- make it positive at the expense of the integer part.
+        -- Code below expects that "nsec" value is always positive.
         if fraction < 0 then
             sec_int = sec_int - 1
-            fraction = 1 + fraction
+            fraction = fraction + 1
         end
```

Reply:
Thanks!